examples/dmd/pore_collapse.cpp

```diff
@@ -0,0 +1,299 @@
+                   "Location of training and testing data list.");
+    args.AddOption(&data_dir, "-data", "--data_dir",
+                   "Location of training and testing data.");
+    args.AddOption(&var_name, "-var", "--var_name",
+                   "Name of variable.");
+    args.Parse();
+    if (!args.Good())
+    {
+        if (mpi.Root())
+        {
+            args.PrintUsage(cout);
+        }
+        return 1;
+    }
+    if (mpi.Root())
+    {
+        args.PrintOptions(cout);
+    }
+
+    CAROM::CSVDatabase* csv_db(new CAROM::CSVDatabase);
+
+    std::string variable = std::string(var_name);
+    int nelements = -1;
+    csv_db->getIntegerArray(std::string(data_dir) + "/dim.txt", &nelements, 1);
+    if (myid == 0)
+    {
+        cout << "Variable " << var_name << " has dimension " << nelements << "." << endl;
+    }
+
+    int dim = nelements;
+    std::vector<int> idx_state;
+    csv_db->getIntegerArray(std::string(data_dir) + "/index.csv", idx_state, false);
+    if (idx_state.size() > 0)
+    {
+        dim = idx_state.size();
+        cout << "Restricting on " << dim << " entries out of " << nelements << "." << endl;
+        for (int k = 0; k < dim; ++k) cout << k << "-th index: " << idx_state[k] << endl;
+    }
+
+    CAROM::DMD* dmd = nullptr;
+    CAROM::AdaptiveDMD* admd = nullptr;
+    if (dtc > 0.0){
+        dmd = new CAROM::DMD(dim, dtc);
+    }
+    else
+    {
+        dmd = new CAROM::AdaptiveDMD(dim, ddt, "LS", "G", dmd_epsilon);
+        admd = dynamic_cast<CAROM::AdaptiveDMD*> (dmd);
+    }
+
+    std::vector<std::string> training_par_list;
+    csv_db->getStringList(std::string(list_dir) + "/training_gpa", training_par_list, false);
+    int npar = training_par_list.size();
+
+    double* sample = new double[dim];
+    StopWatch dmd_training_timer, dmd_prediction_timer;
+    dmd_training_timer.Start();
+
+    for (int idx_test = 0; idx_test < npar; ++idx_test) 
+    {
+        std::string par_dir = training_par_list[idx_test]; // *gpa
+        if (myid == 0)
+        {
+            cout << "Loading samples for " << par_dir << " to train DMD." << endl;
+        }
+        std::vector<std::string> snap_list;
+        csv_db->getStringList(std::string(list_dir) + "/" + par_dir, snap_list, false);
+        int num_snap = snap_list.size();
+        double tval = 0.0;
+        for (int idx_snap = 0; idx_snap < num_snap; ++idx_snap)
+        {
+            std::string snap = snap_list[idx_snap]; // run_036.*
+            csv_db->getDoubleArray(std::string(data_dir) + "/" + par_dir + "/" + snap + "/tval.txt", &tval, 1);
+            if (idx_snap == 0)
+            {
+                t_init = tval;
+            }
+            std::string data_filename = std::string(data_dir) + "/" + par_dir + "/" + snap + "/" + variable + ".csv"; // {zone_*,tkelv}.csv
+            csv_db->getDoubleArray(data_filename, sample, nelements, idx_state);
+            dmd->takeSample(sample, tval - t_init);
```

Comment:
I implemented a time offset in AdaptiveDMD. I believe this can be changed to just "tval". d_t_offset in AdaptiveDMD will be set to t_init and will do the subtraction for yo.

Reply:
Got it. Thanks @kevinhkhuynh